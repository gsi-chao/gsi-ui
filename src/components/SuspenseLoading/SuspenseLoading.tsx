import React, { Suspense } from 'react';
import { Intent } from '@blueprintjs/core';
import { VSpinner } from '../Spinner';

interface VSuspenseReactProps {
  children: any;
  intent?: Intent;
}

export const VSuspenseLoading = (props: VSuspenseReactProps) => {
  return (
    <Suspense fallback={<VSpinner intent={props.intent} />}>
      {props.children}
    </Suspense>
  );
};
