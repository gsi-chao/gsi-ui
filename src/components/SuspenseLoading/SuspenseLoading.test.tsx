import React from 'react';
import renderer from 'react-test-renderer';
import { VSuspenseLoading } from './SuspenseLoading';

it('render correctly Suspense Loading', () => {
  const TextInputComponent = renderer
    .create(
      <VSuspenseLoading>
        <div />
      </VSuspenseLoading>
    )
    .toJSON();
  expect(TextInputComponent).toMatchSnapshot();
});
