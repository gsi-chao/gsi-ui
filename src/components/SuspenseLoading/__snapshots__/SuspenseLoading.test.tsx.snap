// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render correctly Suspense Loading 1`] = `<div />`;
